src/transport/GroupPeerMessageCounter.cpp

```diff
@@ -0,0 +1,280 @@
+                mGroupFabrics[it].mControlPeerCount++;
+            }
+            else
+            {
+                mGroupFabrics[it].mDataGroupSenders[0].mNodeId = nodeId;
+                counter                                        = &(mGroupFabrics[it].mDataGroupSenders[0].msgCounter);
+                mGroupFabrics[it].mDataPeerCount++;
+            }
+            return CHIP_NO_ERROR;
+        }
+
+        if (fabricId == mGroupFabrics[it].mFabricId)
+        {
+            if (isControl)
+            {
+                for (uint32_t nodeIt = 0; nodeIt < GROUP_MSG_COUNTER_MAX_NUMBER_OF_GROUP_CONTROL_PEER; nodeIt++)
+                {
+                    if (mGroupFabrics[it].mControlGroupSenders[nodeIt].mNodeId == kUndefinedNodeId)
+                    {
+                        // Already iterate through all known NodeId
+                        // Add the new peer to save some processing time
+                        mGroupFabrics[it].mControlGroupSenders[nodeIt].mNodeId = nodeId;
+                        counter = &(mGroupFabrics[it].mControlGroupSenders[nodeIt].msgCounter);
+                        mGroupFabrics[it].mControlPeerCount++;
+                        return CHIP_NO_ERROR;
+                    }
+
+                    if (mGroupFabrics[it].mControlGroupSenders[nodeIt].mNodeId == nodeId)
+                    {
+                        counter = &(mGroupFabrics[it].mControlGroupSenders[nodeIt].msgCounter);
+                        return CHIP_NO_ERROR;
+                    }
+                }
+            }
+            else
+            {
+                for (uint32_t nodeIt = 0; nodeIt < GROUP_MSG_COUNTER_MAX_NUMBER_OF_GROUP_DATA_PEER; nodeIt++)
+                {
+                    if (mGroupFabrics[it].mDataGroupSenders[nodeIt].mNodeId == kUndefinedNodeId)
+                    {
+                        // Already iterate through all known NodeId
+                        // Add the new peer to save some processing time
+                        mGroupFabrics[it].mDataGroupSenders[nodeIt].mNodeId = nodeId;
+                        counter = &(mGroupFabrics[it].mDataGroupSenders[nodeIt].msgCounter);
+                        mGroupFabrics[it].mDataPeerCount++;
+                        return CHIP_NO_ERROR;
+                    }
+
+                    if (mGroupFabrics[it].mDataGroupSenders[nodeIt].mNodeId == nodeId)
+                    {
+                        counter = &(mGroupFabrics[it].mDataGroupSenders[nodeIt].msgCounter);
+                        return CHIP_NO_ERROR;
+                    }
+                }
+            }
+            // Exceed the Max number of Group peer
+            return CHIP_ERROR_TOO_MANY_PEER_NODES;
+        }
+    }
+
+    // Exceed the Max number of Group peer
+    return CHIP_ERROR_TOO_MANY_PEER_NODES;
+}
+
+// Used in case of MCSP failure
+CHIP_ERROR GroupPeerTable::RemovePeer(FabricId fabricId, NodeId nodeId, bool isControl)
```

Comment:
This is ambiguous.  This interface should be (fabricIndex, nodeId, isControl), or (roortCAPubKey, fabricId, nodeId, isControl).

Reply:
Confusion between FabricID and FabricIndex, will change.

Reply:
Done